spec/requests/uploads_controller_spec.rb

```diff
@@ -774,7 +778,577 @@ def upload_file(file, folder = "images")
+          params: {
+            file_name: "test.zip",
+            file_size: 9999999,
+            upload_type: "composer",
+            content_type: "application/zip"
+          }
+        }
+        expect(response.status).to eq(422)
+        expect(response.body).to include(I18n.t("upload.attachments.too_large", max_size_kb: SiteSetting.max_attachment_size_kb))
+      end
+
+      def stub_create_multipart_request
+        create_multipart_result = <<~BODY
+        <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n
+        <InitiateMultipartUploadResult>
+           <Bucket>s3-upload-bucket</Bucket>
+           <Key>uploads/default/test_0/temp/28fccf8259bbe75b873a2bd2564b778c/test.png</Key>
+           <UploadId>#{mock_multipart_upload_id}</UploadId>
+        </InitiateMultipartUploadResult>
+        BODY
+        stub_request(
+          :post,
+          "https://s3-upload-bucket.s3.us-west-1.amazonaws.com/uploads/default/test_0/temp/28fccf8259bbe75b873a2bd2564b778c/test.png?uploads"
+        ).to_return({ status: 200, body: create_multipart_result })
+      end
+
+      it "creates a multipart upload and creates an external upload stub that is marked as multipart" do
+        stub_create_multipart_request
+        post "/uploads/create-multipart.json", {
+          params: {
+            file_name: "test.png",
+            file_size: 1024,
+            upload_type: "composer",
+            content_type: "image/png"
+          }
+        }
+
+        expect(response.status).to eq(200)
+        result = response.parsed_body
+
+        external_upload_stub = ExternalUploadStub.where(
+          unique_identifier: result["unique_identifier"],
+          original_filename: "test.png",
+          created_by: user,
+          upload_type: "composer",
+          key: result["key"],
+          external_upload_identifier: mock_multipart_upload_id,
+          multipart: true,
+          filesize: 1024
+        )
+        expect(external_upload_stub.exists?).to eq(true)
+        expect(result["key"]).to include(FileStore::S3Store::TEMPORARY_UPLOAD_PREFIX)
+        expect(result["external_upload_identifier"]).to eq(mock_multipart_upload_id)
+        expect(result["key"]).to eq(external_upload_stub.last.key)
+      end
+
+      it "rate limits" do
+        RateLimiter.enable
+        RateLimiter.clear_all!
+
+        stub_create_multipart_request
+        stub_const(UploadsController, "CREATE_MULTIPART_RATE_LIMIT_PER_MINUTE", 1) do
+          post "/uploads/create-multipart.json", params: {
+            file_name: "test.png",
+            upload_type: "composer",
+            content_type: "image/png",
+            file_size: 1024
+          }
+          post "/uploads/create-multipart.json", params: {
+            file_name: "test.png",
+            upload_type: "composer",
+            content_type: "image/png",
+            file_size: 1024
+          }
+        end
+        expect(response.status).to eq(429)
```

Comment:
```suggestion
        stub_const(UploadsController, "CREATE_MULTIPART_RATE_LIMIT_PER_MINUTE", 1) do
          post "/uploads/create-multipart.json", params: {
            file_name: "test.png",
            upload_type: "composer",
            content_type: "image/png",
            file_size: 1024
          }

          expect(response.status).to eq(200)
          
          post "/uploads/create-multipart.json", params: {
            file_name: "test.png",
            upload_type: "composer",
            content_type: "image/png",
            file_size: 1024
          }

          expect(response.status).to eq(429)
        end
```

Reply:
This is such a weird diff that github has dreamed up for a two line change :sweat_smile: